optimum/onnxruntime/modeling_ort.py

```diff
@@ -852,6 +863,39 @@ def raise_on_numpy_input_io_binding(self, use_torch: bool):
                 " with model.use_io_binding = False, or pass torch.Tensor inputs instead."
             )
 
+    def _prepare_onnx_inputs(
```

Comment:
it only applies when needed so for performance I think there's no added overhead vs what used to be done.
where things can be optimized and i think can be viewed as the optimization oriented path, is i/o binding:
- pre-allocation of output buffers, not during forward but before that, and dynamically augmenting the size of output buffers when batch size changes.
- decoder models with i/o binding where static cache implementation can be used as output buffers to reduce the overhead of their creation.
- decoder models input/output synchronization which can be before and after the generation loop instead of each forward call (if that's possible).

Reply:
I kind of agree, just not sure what each ORTModelForXXX is for then